app/controllers/declaration/DeclarationHolderController.scala

```diff
@@ -43,10 +43,11 @@ class DeclarationHolderController @Inject()(
   import DeclarationHolderController._
 
   def displayPage(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
-    val frm = addAnotherYesNoForm.withSubmissionErrors()
-    request.cacheModel.parties.declarationHoldersData match {
-      case Some(data) if data.holders.nonEmpty => Ok(declarationHolderPage(mode, frm, data.holders))
-      case _                                   => navigator.continueTo(mode, routes.DeclarationHolderAddController.displayPage)
+    val holders = request.cacheModel.parties.declarationHoldersData.map(_.holders).getOrElse(Seq.empty)
+    if (holders.isEmpty) navigator.continueTo(mode, routes.DeclarationHolderRequiredController.displayPage)
+    else {
+      val frm = addAnotherYesNoForm.withSubmissionErrors()
+      Ok(declarationHolderPage(mode, frm, holders))
```

Comment:
To my eyes this is harder to read than the original match statement version.

Reply:
I would agree if `frm` were used by both case branches, but IMO in this case, moving `frm` to the branch where it is actually used, the resulting `match` would be more convoluted and verbose
```
request.cacheModel.parties.declarationHoldersData match {
  case Some(data) if data.holders.nonEmpty =>
    val frm = addAnotherYesNoForm.withSubmissionErrors()
    Ok(declarationHolderPage(mode, frm, data.holders))

  case _  => navigator.continueTo(mode, routes.DeclarationHolderAddController.displayPage)
}

```

Reply:
How about

```
request.cacheModel.parties.declarationHoldersData match {
  case Some(data) if data.holders.nonEmpty =>
    Ok(declarationHolderPage(mode, addAnotherYesNoForm.withSubmissionErrors(), data.holders))
  case _  => navigator.continueTo(mode, routes.DeclarationHolderAddController.displayPage)
}
```

Reply:
I just pushed a slight improvement but if you both still like more the `match` block here I'll change it accordingly.

Reply:
Personally I prefer the match version as it takes less effort to reason about :slightly_smiling_face:

Reply:
Even after my last changes? :open_mouth:

Reply:
i'm less bothered now. nice improvement.